### DISCUSSION TITLE
How to configure ComfyUi to run with attributes --no-half --skip-torch-cuda-test --upcast-sampling --no-half-vae --use-cpu interrogate

### DISCUSSION THREAD

**Discussion by RichEdg** [comfyanonymous/ComfyUI#1008] [Q&A]
I have to instillations of ComfyUI on myMacBook Pro M2Max. The first was installed using the ComfyUI_windows_portable install and it runs through Automatic1111. I had installed the ComfyUI extension in Automatic1111 and was running it within Automatic1111. I noticed that in the Terminal window you could launch the ComfyUI directly in a browser with a URL link. So that is how I was running ComfyUI. I would generate an image using SDXL base and refiner models at 1080x1080 in 240 seconds. So that was not too bad! I could even use a workflow that output at 8k. But I wanted to have a standalone version of ComfyUI. After much research, some help from a few kind people on Reddit, and using ChatGPT to answer questions,  I finally got it set up and running. But the speed was pathetic. VERY slow. Well over 30 minutes for a generation.

Why would two version of same program run at such different speeds? I looked at the how Automatic1111 starts up. It runs with the following attributes to speed things up on a mac:  --no-half --skip-torch-cuda-test --upcast-sampling --no-half-vae --use-cpu interrogate. I do not know which of these is essential for the speed up process. But is there a way of configuring the command that I use to start the ComfyUI: python main.py so that it uses a similar configuration to allow for a much speedier image generation time??? When I looked in the file main.py I could see no obvious way to modify it to allow the attributes to be set. 

Can anyone give me some guidance please on how to do this? Is this even a thing?

Thank you for any help/suggestions!

**Comment by zoemax** [comment]
if you cd into your comfy directory, just run: python main.py -h. - or python3 main.py -h. -- and you'll should see arguments that can be passed on the command line: 

ie: ~/ComfyUI| python3 main.py --listen --use-split-cross-attention

See: [This](https://github.com/comfyanonymous/ComfyUI/issues/1029)
